catalog/spatial/registry/registry-publication-update-handler/src/main/java/org/codice/ddf/registry/publication/RegistryPublicationHandler.java

```diff
@@ -0,0 +1,108 @@
+
+import org.apache.commons.collections.CollectionUtils;
+import org.codice.ddf.registry.common.RegistryConstants;
+import org.codice.ddf.registry.common.metacard.RegistryObjectMetacardType;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminException;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminService;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+
+public class RegistryPublicationHandler implements EventHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RegistryPublicationHandler.class);
+
+    private static final String METACARD_PROPERTY = "ddf.catalog.event.metacard";
+
+    private FederationAdminService adminService;
+
+    private ExecutorService executor;
+
+    private int threadPoolSize = 1;
+
+    public RegistryPublicationHandler(FederationAdminService adminService) {
+        this.adminService = adminService;
+    }
+
+    public void init() {
+        if (executor == null) {
+            executor = Executors.newFixedThreadPool(threadPoolSize);
+        }
+    }
+
+    public void setExecutorService(ExecutorService service) {
+        executor = service;
+    }
+
+    public void destroy() {
+        executor.shutdown();
+    }
+
+    @Override
+    public void handleEvent(Event event) {
+        Metacard mcard = (Metacard) event.getProperty(METACARD_PROPERTY);
+        if (mcard == null) {
+            return;
+        }
+        if (mcard.getTags()
+                .contains(RegistryConstants.REGISTRY_TAG)) {
+            executor.execute(() -> {
+                processUpdate(mcard);
+            });
+        }
+    }
+
+    protected void processUpdate(Metacard mcard) {
+        Attribute publishedLocations =
+                mcard.getAttribute(RegistryObjectMetacardType.PUBLISHED_LOCATIONS);
+        if (publishedLocations != null) {
+            Attribute lastPublished = mcard.getAttribute(RegistryObjectMetacardType.LAST_PUBLISHED);
+            Date datePublished = null;
+            if (lastPublished != null) {
+                datePublished = (Date) lastPublished.getValue();
+            }
+
+            Set<String> locations = new HashSet<>(publishedLocations.getValues()).stream()
+                    .map(e -> e.toString())
+                    .collect(Collectors.toSet());
+
+            if ((datePublished == null || datePublished.before(mcard.getModifiedDate()))
+                    && CollectionUtils.isNotEmpty(locations)) {
+                try {
+                    adminService.updateRegistryEntry(mcard, locations);
+                    mcard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.LAST_PUBLISHED,
+                            mcard.getModifiedDate()));
+                    adminService.updateRegistryEntry(mcard);
```

Comment:
Any reason we need to call `updateRegistry` again? Couldn't we update the modified date first and call it only once?

If it has to be done this way, are those methods called by any other threads outside of this class? Any chance that the second call may clobber changes made by somebody else?